jobserver/github.py

```diff
@@ -21,17 +21,42 @@
     "User-Agent": "OpenSAFELY Jobs",
 }
 
+# Clients should catch GitHubError to handle common, often transient, connection
+# issues gracefully. Some HTTPError status codes indicate specific API errors
+# related to remote state (e.g., attempting to create an object that already
+# exists). These cases should raise specific HTTPError exceptions from this
+# module, allowing clients to handle such errors without relying on internal
+# implementation details.
+
 
 class GitHubError(Exception):
-    """Base exception to target all other exceptions we define here"""
+    """Base exception for this module. A problem contacting or using the GitHub
+    API."""
+
+
+class Timeout(GitHubError):
+    """A request to the GitHub API timed out."""
+
+
+class ConnectionException(GitHubError):
+    """A connection error occurred while contacting the GitHub API."""
+
+    # ConnectionError is a Python default exception class, so let's avoid using
+    # that name. Otherwise we might have mirrored the name of
+    # requests.exceptions.ConnectionError.
```

Comment:
I'd stick with `ConnectionError` and use the module to disambiguate between them and the builtin:

```py
from jobserver import github
import requests

github.ConnectionError
requests.ConnectionError
ConnectionError
```

Otherwise, I'm left wondering how a `ConnectionError` is different to a `ConnectionException`.

Reply:
`requests.ConnectionError` is an implementation detail of this module I don't expect clients to need to know about and that may change in future. Within this module I think the current situation is clear from the code and this comment.